### ISSUE TITLE
eosio::asset::print does not support negative amounts

### ISSUE THREAD

**Issue by arhag** [EOSIO/eosio.cdt#230]
Trying to print an asset with a negative `amount` will result in garbage characters in the string after the decimal point.

**Comment by larryk85** [comment]
Resolved by PR #233

**Comment by bytemaster** [comment]
When printing negative assets it displays a positive number.

**Comment by jeffreyssmith2nd** [comment]
The current behavior I'm seeing (on 1.6.2) is this code produces the following output. Seems like whole negative numbers are handled correctly but fractional numbers are not.

```
#include <eosio/eosio.hpp>
#include <eosio/asset.hpp>
#include <eosio/symbol.hpp>
#include <eosio/print.hpp>

using namespace eosio;

class assettest : public contract {
    public:
        using contract::contract;

        [[eosio::action]]
        void test() {
            std::string sym = "SYS";
            eosio::asset a(-40000, eosio::symbol("SYS", 4));
            eosio::asset b(-4000, eosio::symbol("SYS", 4));
            eosio::asset c(-400, eosio::symbol("SYS", 4));
            eosio::asset d(-40, eosio::symbol("SYS", 4));
            eosio::asset e(-4, eosio::symbol("SYS", 4));
            print(a);
            print(" ");
            print(b);
            print(" ");
            print(c);
            print(" ");
            print(d);
            print(" ");
            print(e);
            print(" ");
        }
};
```

```
-4.0000 SYS 0.4000 SYS 0.0400 SYS 0.0040 SYS 0.0004
```

**Comment by jeffreyssmith2nd** [comment]
Fixed in #557
